types/geojson/index.d.ts

```diff
@@ -70,67 +76,80 @@ export interface GeometryObject extends GeoJsonObject {
     type: GeoJsonGeometryTypes;
 }
 
+/**
+ * Union of geometry objects.
+ * https://tools.ietf.org/html/rfc7946#section-3
+ */
+export type Geometry = Point | MultiPoint | LineString | Polygon | MultiPolygon | GeometryCollection;
+
 /**
  * Point geometry object.
  * https://tools.ietf.org/html/rfc7946#section-3.1.2
  */
-export interface Point extends GeometryObject {
+export interface Point {
```

Comment:
I disagree to stop extending from `GeometryObject`. As you can see in https://tools.ietf.org/html/rfc7946 GeoJSON is structured the other way. You have a definition for a GeoJSONObject that includes for example the `bbox` property. Then you have the GeometryObject that extends from a GeoJSONObject. Then you have for example a `Point` which extends from a GeometryObject.

Reply:
When you `extend` from `GeometryObject` it breaks the discriminated unions. TS is structurally typed, so `Point` implicitly satisfies the `GeometryObject`. That means no existing code will break as a result of this change.

Reply:
I tested again, and it looks like you're right. Extending `GeometryObject` doesn't break it.